app/src/lib/git/cherry-pick.ts

```diff
@@ -0,0 +1,47 @@
+import { Repository } from '../../models/repository'
+import { git, IGitResult } from './core'
+
+/** The app-specific results from attempting to cherry pick commits*/
+export enum CherryPickResult {
+  /**
+   * Git completed the cherry pick without reporting any errors, and the caller can
+   * signal success to the user.
+   */
+  CompletedWithoutError = 'CompletedWithoutError',
+
+  /**
+   * An unexpected error as part of the cherry pick flow was caught and handled.
+   *
+   * Check the logs to find the relevant Git details.
+   */
+  Error = 'Error',
+}
+
+/**
+ * A stub function to initiate cherry picking in the app.
+ *
+ * @param revisionRange - this could be a single commit sha or could be a range
+ * of commits like sha1..sha2 or inclusively sha1^..sha2
+ */
+export async function cherryPick(
+  repository: Repository,
+  revisionRange: string
+): Promise<CherryPickResult> {
+  const result = await git(
+    ['cherry-pick', revisionRange],
+    repository.path,
+    'cherry pick'
+  )
+
+  return parseCherryPickResult(result)
+}
+
+function parseCherryPickResult(result: IGitResult): CherryPickResult {
```

Comment:
Thanks for the reminder! I  believe that may be more prudent when I add the progress callback in - there is a bit more parsing on that - it pulls a few more attributes out to monitor progress. All this method is responsible for is mapping the dugite/git exit/error code to a cherry pick result (following the pattern established in rebase) to allow for the app store to translate that into a user friendly response. But, I will keep it in mind in case there is better way to do both now :)